org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/internal/ResolvedTypes.java

```diff
@@ -159,11 +158,18 @@ public static class Shared {
 		final IFeatureScopeTracker featureScopeTracker;
 		final IssueSeverities issueSeverities;
 		
-		final Set<JvmIdentifiableElement> allTypes = Sets.newHashSet();
-		final Set<JvmIdentifiableElement> allReassignedTypes = Sets.newHashSet();
-		final Set<XExpression> allExpressionTypes = Sets.newHashSet();
-		final Set<XExpression> allLinking = Sets.newHashSet();
-		final Set<Object> allResolvedTypeParameters = Sets.newHashSet();
+		final Set<JvmIdentifiableElement> allTypes = new HashSet<>();
```

Comment:
order does not matter here?

Reply:
I basically inlined newHashSet to avoid the array creation for the var-args. Order wasn't relevant here in the past and shouldn't be in the foreseeable future.